deeplake/core/chunk/chunk_compressed_chunk.py

```diff
@@ -35,16 +36,194 @@ def __init__(self, *args, **kwargs):
 
-    def extend_if_has_space(self, incoming_samples: List[InputSample], update_tensor_meta: bool = True) -> float:  # type: ignore
+    def extend_if_has_space(self, incoming_samples: List[InputSample], update_tensor_meta: bool = True, lengths: Optional[List[int]] = None) -> float:  # type: ignore
         self.prepare_for_write()
+        if lengths is not None:  # this is triggered only for htype == "text"
+            return self.extend_if_has_space_byte_compression_text(
+                incoming_samples, update_tensor_meta, lengths
+            )
         if self.is_byte_compression:
+            if isinstance(incoming_samples, np.ndarray):
+                return self.extend_if_has_space_byte_compression_numpy(
+                    incoming_samples, update_tensor_meta
+                )
             return self.extend_if_has_space_byte_compression(
                 incoming_samples, update_tensor_meta=update_tensor_meta
             )
         return self.extend_if_has_space_image_compression(
             incoming_samples, update_tensor_meta=update_tensor_meta
         )
 
+    def extend_if_has_space_byte_compression_text(
+        self,
+        incoming_samples: np.ndarray,
+        update_tensor_meta: bool = True,
+        lengths: Optional[List[int]] = None,
+    ):
+        sample_nbytes = np.mean(lengths)
+        min_chunk_size = self.min_chunk_size
+        decompressed_bytes = self.decompressed_bytes
+        while True:
+            if sample_nbytes:
+                num_samples = int(
+                    max(
+                        0,
+                        min(
+                            len(incoming_samples),
+                            (
+                                (min_chunk_size / self._compression_ratio)
+                                - len(decompressed_bytes)
+                            )
+                            // sample_nbytes,
+                        ),
+                    )
+                )
+            else:
+                num_samples = len(incoming_samples)
+            if not num_samples:
+
+                # Check if compression ratio is actually better
+                s = self._text_sample_to_byte_string(incoming_samples[0])
+                new_decompressed = decompressed_bytes + s
+                compressed_bytes = compress_bytes(
+                    new_decompressed, compression=self.compression
+                )
+
+                if len(compressed_bytes) <= min_chunk_size:
+                    self._compression_ratio /= 2
+                    continue
+                else:
+                    if self.decompressed_bytes:
+                        break
+                    else:
+                        self.decompressed_bytes = new_decompressed
+                        self._data_bytes = compressed_bytes
+                        self._changed = False
+                        num_samples = 1
+                        lengths[0] = len(s)
+                        break
```

Comment:
Since we dont have tiling for text, we give up and append the first sample as such to the chunk.